rosbag2_cpp/src/rosbag2_cpp/info.cpp

```diff
@@ -14,16 +14,23 @@
 
 #include "rosbag2_cpp/info.hpp"
 
-#include <memory>
+#include <unordered_map>
+#include <unordered_set>
 #include <stdexcept>
 #include <string>
 
+#include "rcl/service_introspection.h"
+#include "rmw/rmw.h"
+
 #include "rcpputils/filesystem_helper.hpp"
+#include "rosbag2_cpp/service_utils.hpp"
 #include "rosbag2_storage/logging.hpp"
 #include "rosbag2_storage/metadata_io.hpp"
 #include "rosbag2_storage/storage_interfaces/read_only_interface.hpp"
 #include "rosbag2_storage/storage_factory.hpp"
 
+#include "service_msgs/msg/service_event_info.hpp"
+
```

Comment:
Comments from MichaelOrlov

```suggestion
#include "rmw/rmw.h"
#include "rosidl_typesupport_cpp/message_type_support.hpp"
#include "rcpputils/filesystem_helper.hpp"
#include "service_msgs/msg/service_event_info.hpp"

#include "rosbag2_cpp/service_utils.hpp"
#include "rosbag2_storage/metadata_io.hpp"
#include "rosbag2_storage/storage_factory.hpp"
```

Reply:
Updated